### DISCUSSION TITLE
Jenkins setup script issue in version 5.2.0

### DISCUSSION THREAD

**Discussion by gohilamariappan** [Sunbird-Ed/Community#513] [Installation] [answered]
Hi Team!

In ubuntu 20 server , Jenkins setup script is giving the below error , while installing jenkins.
 Kindly check and suggest.

############################
W: GPG error: https://pkg.jenkins.io/debian-stable binary/ Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 
E: The repository 'https://pkg.jenkins.io/debian-stable binary/ Release' is not signed.
##########

Regards,
Gohila

**Comment by tsprasath** [comment]
@gohilamariappan  Please check this command https://github.com/project-sunbird/sunbird-devops/blob/release-5.1.0/deploy/jenkins/jenkins-server-setup.sh#LL15C1-L16C70 and share the logs

**Reply by heungheung** [reply]
@tsprasath 

https://www.jenkins.io/blog/2023/03/27/repository-signing-keys-changing/

they will need to do something like
```
curl -fsSL https://pkg.jenkins.io/debian-stable/jenkins.io-2023.key | sudo tee \
  /usr/share/keyrings/jenkins-keyring.asc > /dev/null
echo deb [signed-by=/usr/share/keyrings/jenkins-keyring.asc] \
  https://pkg.jenkins.io/debian-stable binary/ | sudo tee \
  /etc/apt/sources.list.d/jenkins.list > /dev/null
```

**Comment by gohilamariappan** [comment, marked as answer]
Hi @tsprasath 

I have ran the below command. 
sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 5BA31D57EF5975CA


Regards,
Gohila

**Comment by Krishnaj20** [comment]
@chitranshu-keshav - is this thread resolved?

**Reply by santhosh-tg** [reply]
Hi @gohilamariappan  please close the thread if the issue is resolved. Also, if you have implemented the solution suggested by @heungheung  please feel free to contribute back by raising a PR

**Comment by gohilamariappan** [comment]
Issue resolved.

Regards,
Gohila
